exampleLitActions/src/conditionalSigning.ts

```diff
@@ -0,0 +1,238 @@
+import { LitNodeClient } from '@lit-protocol/lit-node-client';
+import { LIT_RPC, LIT_NETWORK, LIT_ABILITY } from "@lit-protocol/constants";
+import {
+  createSiweMessage,
+  generateAuthSig,
+  LitActionResource,
+} from "@lit-protocol/auth-helpers";
+import { LitContracts } from "@lit-protocol/contracts-sdk";
+import * as ethers from "ethers";
+
+// Utility function to get environment variables
+const getEnv = (name: string): string => {
+  const value = process.env[name];
+  if (!value) {
+    console.warn(`Environment variable ${name} is not set`);
+  }
+  return value || "";
+};
+
+// Environment variables
+const ETHEREUM_PRIVATE_KEY = getEnv("ETHEREUM_PRIVATE_KEY");
+const CHAIN_TO_CHECK_CONDITION_ON = getEnv("CHAIN_TO_CHECK_CONDITION_ON") || "ethereum";
+const LIT_PKP_PUBLIC_KEY = process.env["LIT_PKP_PUBLIC_KEY"];
+const LIT_CAPACITY_CREDIT_TOKEN_ID = process.env["LIT_CAPACITY_CREDIT_TOKEN_ID"];
+
+// Define the Lit Action code for conditional signing
+const litActionCode = `
+async () =>  {
+  try {
+    // test an access control condition
+    const testResult = await Lit.Actions.checkConditions({
+      conditions,
+      authSig,
+      chain,
+    });
+
+    if (!testResult) {
+      LitActions.setResponse({ response: "address does not have 1 or more Wei on Ethereum Mainnet" });
+      return;
+    }
+
+    const sigShare = await LitActions.signEcdsa({
+      toSign: dataToSign,
+      publicKey,
+      sigName: "sig",
+    });
+  } catch (error) {
+    LitActions.setResponse({ response: error.message });
+  }
+};
+`;
+
+// Result interface for our function
+interface ConditionalSigningResult {
+  success: boolean;
+  signatureResult?: Record<string, any>;
+  response?: string;
+  error?: string;
+}
```

Comment:
Unless you feel strongly otherwise, we should return a custom type from the example functions, we should just return the raw Lit SDK response and then handle the parsing inside the test. We don't want to clutter the code example with code that's not technically required - it can become confusing for the user copy/pasting the example